Services/DeviceModels.cs

```diff
@@ -160,7 +167,17 @@ public class DeviceModels : IDeviceModels
         /// </summary>
         public async Task DeleteAsync(string id)
         {
+            if (this.CheckDeviceModelExistence(id))
+            {
+                throw new UnauthorizedAccessException("Cannot delete a stock device model");
+            }
+
             await this.customDeviceModels.DeleteAsync(id);
         }
+
+        private bool CheckDeviceModelExistence(string id)
+        {
+            return  this.stockDeviceModels.GetList().Any(model => model.Id == id);
```

Comment:
better if the comparison is not case sensitive

Reply:
Done.